### PR TITLE
in the constructor WebScraper domain validation has been added.

### PR DESCRIPTION
**Pull request by v-grabko1999**
I use google translator.

I added a baseURL validation check, created descriptions of validation errors that turn into exceptions in the WebScraper constructor.

### CODE REVIEW

File: lib/validation.dart [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        return false;
+      }
+    }
+    return true;
+  }
+}
+
+class ValidationRet {
+  bool Is;
+  String Description;
```
Comment (tusharojha) [inline lib/validation.dart:118]: use camel-casing for class variables.

File: test/validation_test.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import 'package:test/test.dart';
+import 'package:web_scraper/validation.dart';
+
+void main() {
+  test('Error: use [http/https] protocol', () {
```
Comment (tusharojha) [inline test/validation_test.dart:5]: add positive test messages for all tests in the file.
example: Checks for [http/https] protocol

File: lib/validation.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
+library web_scraper;
+
+class Validation {
+  static RegExp _ipv6 =
+      RegExp(r'^::|^::1|^([a-fA-F0-9]{1,4}::?){1,7}([a-fA-F0-9]{1,4})$');
+  static RegExp _ipv4Maybe =
+      new RegExp(r'^(\d?\d?\d)\.(\d?\d?\d)\.(\d?\d?\d)\.(\d?\d?\d)$');
+
+  //
```
Comment (tusharojha) [inline lib/validation.dart:9]: Remove this empty comment.

File: lib/validation.dart [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          part[part.length - 1] == '-' ||
+          part.indexOf('---') >= 0) {
+        return false;
+      }
+    }
+    return true;
+  }
+}
+
+class ValidationRet {
```
Comment (tusharojha) [inline lib/validation.dart:116]: change ValidationRet to something more understandable / related.

**Comment by tusharojha** [comment]
Also, As you know Travis CI is not yet set up. I request you to format all dart files using the `flutter format` command.

**Comment by v-grabko1999** [comment]
I will work on this in a few days.

**Comment by tusharojha** [comment]
Merging this PR as the source repository has been deleted.
@v-grabko1999 No issue I will make the suggested changes.

**Pull request merged**